### DISCUSSION TITLE
groth16.Verify Resulting In sigpanic: panicmem

### DISCUSSION THREAD

**Discussion by Kasshern** [Consensys/gnark#274] [Q&A] [answered]
When running  

`func CallCubicCircuit() string {
	var zkcircuit Circuit
	r1cs, err := frontend.Compile(ecc.BN254, backend.GROTH16, &zkcircuit)
	if err != nil {
		log.Fatal(err)
	}

	// generating pk, vk
	assignment := &Circuit{
		X: 42,
		Y: 42,
	}

	witness, _ := frontend.NewWitness(assignment, ecc.BN254)
	publicWitness, err := witness.Public()
	pk, vk, err := groth16.Setup(r1cs)
	proof, err := groth16.Prove(r1cs, pk, witness)
	err = groth16.Verify(proof, vk, publicWitness)
	if err != nil {
		return "INVALID PROOF!"
	} else {
		return "VALID PROOF"
	}`

I get the following Stack Trace 

C:/Program Files/Go/src/runtime/panic.go:221 (0x7c9b9c)
        panicmem: panic(memoryError)
C:/Program Files/Go/src/runtime/signal_windows.go:254 (0x7c9b6c)
        sigpanic: panicmem()
C:/Users/.../go/pkg/mod/github.com/consensys/gnark@v0.6.4/internal/backend/bn254/groth16/prove.go:47 (0xcbac19)
        (*Proof).isValid: return proof.Ar.IsInSubGroup() && proof.Krs.IsInSubGroup() && proof.Bs.IsInSubGroup()
C:/Users/.../go/pkg/mod/github.com/consensys/gnark@v0.6.4/internal/backend/bn254/groth16/verify.go:45 (0xcc0ad5)
        Verify: if !proof.isValid() {
C:/Users/.../go/pkg/mod/github.com/consensys/gnark@v0.6.4/backend/groth16/groth16.go:137 (0xcdb184)
        Verify: return groth16_bn254.Verify(_proof, vk.(*groth16_bn254.VerifyingKey), *w)
C:/Users/.../zkcircuits/cubic/cubic.go:46 (0xcdc164)
        CallCubicCircuit: err = groth16.Verify(proof, vk, publicWitness)
C:/Users/.../zkcircuits/main.go:35 (0xe32407)

if X and Y are the correct values the proof returns valid as expected 

Anyone know whats up with this invalid memory address or nil pointer deference?

**Comment by gbotrel** [comment, marked as answer]
hi -- from the snippet, it seems you're not checking the Prover error; 

```
proof, err := groth16.Prove(r1cs, pk, witness)

// missing error check here 

err = groth16.Verify(proof, vk, publicWitness)
if err != nil {
	return "INVALID PROOF!"
} else {
	return "VALID PROOF"
}
```

**Reply by Kasshern** [reply]
Thank you works like a charm now.
